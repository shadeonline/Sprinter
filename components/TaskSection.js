import React, { useState } from 'react';
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';
import TaskCard from '../components/TaskCard';
import { MaterialIcons } from '@expo/vector-icons';

const TaskSection = ({ title, tasks }) => {
  const [isExpanded, setIsExpanded] = useState(true);

  const toggleSection = () => {
    setIsExpanded(!isExpanded);
  };

  return (
    <View>
      <View style={styles.header}>
        <Text style={styles.sectionTitle}>{title}</Text>
        <TouchableOpacity onPress={toggleSection}>
          {isExpanded ? (
            <MaterialIcons name="expand-less" size={24} color="white" />
          ) : (
            <MaterialIcons name="expand-more" size={24} color="white" />
          )}
        </TouchableOpacity>
      </View>
      {isExpanded ? (
        tasks.length > 0 ? (
          tasks.map((task) => (
            <TaskCard key={task.id} task={task} />
          ))
        ) : (
          <Text style={styles.noTasksText}>No tasks</Text>
        )
      ) : null}
    </View>
  );
};

const styles = StyleSheet.create({
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginTop: 25,
  },
  sectionTitle: {
    fontSize: 20,
    fontWeight: 'bold',
    color: 'white',
  },
  noTasksText: {
    fontSize: 16,
    color: 'white',
    textAlign: 'center',
    marginTop: 10,
  },
});

export default TaskSection;
