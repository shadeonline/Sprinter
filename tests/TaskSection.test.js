import React from 'react';
import renderer from 'react-test-renderer';
import TaskSection from '../components/TaskSection';
import { NavigationContainer } from '@react-navigation/native';

// Mock React Navigation's useNavigation
jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      navigate: jest.fn(),
    }),
  };
});

describe('TaskSection', () => {
  it('matches snapshot', () => {
    const tasks = [
      { id: 1, taskTitle: 'Task 1' },
      { id: 2, taskTitle: 'Task 2' },
    ];

    const tree = renderer
      .create(
        <NavigationContainer>
          <TaskSection title="Section Title" tasks={tasks} />
        </NavigationContainer>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
