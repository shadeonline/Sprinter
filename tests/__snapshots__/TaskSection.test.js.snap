// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskSection matches snapshot 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 25,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Section Title
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "white",
        "borderColor": "gray",
        "borderRadius": 8,
        "borderWidth": 3,
        "elevation": 2,
        "margin": 16,
        "opacity": 1,
        "padding": 16,
        "width": "85%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 8,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Task 1
      </Text>
      <View
        style={
          {
            "backgroundColor": "#DCE0E8",
            "borderRadius": 10,
            "color": "gray",
            "fontSize": 14,
            "paddingHorizontal": 20,
            "paddingVertical": 2,
          }
        }
      >
        <Text
          style={
            {
              "color": "gray",
              "fontSize": 14,
            }
          }
        >
          -
        </Text>
      </View>
    </View>
    <Text
      style={
        {
          "color": "gray",
          "fontSize": 14,
        }
      }
    >
      Deadline: 
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "white",
        "borderColor": "gray",
        "borderRadius": 8,
        "borderWidth": 3,
        "elevation": 2,
        "margin": 16,
        "opacity": 1,
        "padding": 16,
        "width": "85%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 8,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Task 2
      </Text>
      <View
        style={
          {
            "backgroundColor": "#DCE0E8",
            "borderRadius": 10,
            "color": "gray",
            "fontSize": 14,
            "paddingHorizontal": 20,
            "paddingVertical": 2,
          }
        }
      >
        <Text
          style={
            {
              "color": "gray",
              "fontSize": 14,
            }
          }
        >
          -
        </Text>
      </View>
    </View>
    <Text
      style={
        {
          "color": "gray",
          "fontSize": 14,
        }
      }
    >
      Deadline: 
    </Text>
  </View>
</View>
`;
